sycl/doc/design/CommandGraph.md

```diff
@@ -224,59 +224,107 @@ there are no parameters to take a wait-list, and the only sync primitive
-and is reset in the prefix. This signal is detected by a new UR return event
-created on UR command-buffer enqueue.
-
-There is also a *WaitEvent* used by the `ur_exp_command_buffer_handle_t` class
-in the prefix to wait on any dependencies passed in the enqueue wait-list.
-This WaitEvent is reset in the suffix.
-
-A command-buffer is expected to be submitted multiple times. Consequently,
-we need to ensure that L0 events associated with graph commands have not
+Zero, the adapter implementation needs extra commands.
+
+* Prefix - Commands added **before** the relevant workload.
+* Suffix - Commands added **after** the relevant workload.
+
+These extra commands operate on L0 event synchronisation primitives, 
+used by the command-list to interact with the external UR wait-list 
+and UR return event required for the enqueue interface.
+Unlike the graph workload (i.e. commands needed to perform the graph workload) 
+the external UR wait-list and UR return event are submission dependent, 
+which mean they can change from one submission to the next.
+
+For performance concerns, the command-list that will execute the graph
+workload is made only once (during the command-buffer finalization stage).
+This allows the adapter to save time when submitting the command-buffer, 
+by executing only this command-list (i.e. without enqueuing any commands 
+of the graph workload).
+
+#### Prefix
+
+The prefix's commands aim to: 
+1. Handle the the list on events to wait on, which is passed by the Run-Time
+when the command-buffer enqueue function is called.
+As mentionned above, this list of events changes from one submission 
+to the next.
+Consequently, managing this mutable command in the graph-workload 
+command-list implies rebuilding the command-list for each submission 
+(note that this can change with mutable command-list).
+To avoid the signifiant time penalty of rebuilding this potentially large 
+command-list each time, we prefer to add the extra command handling the
+wait list into another command-list (*wait command-list*). 
+This command-list consists of a single L0 command: a barrier that waits for 
+ dependencies passed by the wait-list and signals a signal 
+called *WaitEvent* when the barrier is complete.
+This *WaitEvent* is defined in the `ur_exp_command_buffer_handle_t` class.
+In the front of the graph workload command list, an extra barrier command 
+waiting for this event is added (when the command-buffer is created).
+This ensures that the graph workload does not start running before 
+the dependencies to be completed.
+The *WaitEvent* event is reset in the suffix.
+
+
+2. Reset events associated with the command-buffer except the 
+*WaitEvent* event.
+Indeed, L0 events needs to be explicitly reset by an API call 
+(L0 command in our case).
+Since a command-buffer is expected to be submitted multiple times. 
+We need to ensure that L0 events associated with graph commands have not
```

Comment:
```suggestion
Since a command-buffer is expected to be submitted multiple times,
we need to ensure that L0 events associated with graph commands have not
```